src/services/balances.rs

```diff
@@ -2,12 +2,16 @@ use crate::cache::cache_operations::RequestCached;
 use crate::config::{balances_cache_duration, balances_request_timeout};
 use crate::models::backend::balances::Balance as BalanceDto;
 use crate::models::backend::chains::NativeCurrency;
+use crate::models::backend::tokens::{TokenPrice, TokenPriceCore};
 use crate::models::service::balances::{Balance, Balances};
 use crate::providers::fiat::FiatInfoProvider;
 use crate::providers::info::{DefaultInfoProvider, InfoProvider};
 use crate::utils::context::Context;
 use crate::utils::errors::ApiResult;
-use std::cmp::Ordering;
+use rocket::futures::{stream, StreamExt};
+use std::str::FromStr;
+
+const N_CONCURRENT_TOKEN_REQUESTS: usize = 5;
```

Comment:
This should probably be a config/env var

Reply:
I can add it as a environment variable. Didn't do it because I was trying to avoid some "configuration" overload. But for this one might make sense 👍